primefaces/src/main/java/org/primefaces/util/CurrencyValidator.java

```diff
@@ -116,8 +115,9 @@ protected Object parse(String value, DecimalFormat formatter) {
                     buffer.append(pattern.charAt(i));
                 }
             }
-            formatter.applyPattern(buffer.toString());
-            parsedValue = parseFormat(value, formatter);
+            DecimalFormat copyFormatter = (DecimalFormat) formatter.clone();
```

Comment:
Why clone here?

Reply:
It's because the `applyPattern` method is a modifier, so in order to not alter the original decimal format we clone it and modify that instead. That allow the user of CurrencyValidator to reuse the passed DecimalFormat.

Reply:
But wont this be called over and over again on every cell cloning it every time?  Wouldn't it be better to clone it once?